api/helpers/game-states/moves/seven-scuttle/validate.js

```diff
@@ -0,0 +1,82 @@
+const GamePhase = require('../../../../../utils/GamePhase.json');
+
+module.exports = {
+  friendlyName: 'Validate request to seven-scuttle',
+
+  description: 'Verifies whether a request to scuttle with a seven is legal, throwing an explanatory error if not.',
+
+  inputs: {
+    currentState: {
+      type: 'ref',
+      description: 'Object containing the current game state',
+      required: true,
+    },
+    /**
+     * @param { Object } requestedMove - Object describing the request to scuttle from the top of the deck
+     * @param { String } requestedMove.cardId - Card Played for scuttle
+     * @param { String } requestedMove.targetId - Card Targeted for scuttle
+     * @param { MoveType.SEVEN_SCUTTLE } requestedMove.moveType - Specifies that this a sevenScuttle
+     */
+    requestedMove: {
+      type: 'ref',
+      description: 'Object describing the request to scuttle',
+      required: true,
+    },
+    playedBy: {
+      type: 'number',
+      description: 'Player number of player requesting the move',
+      required: true,
+    },
+  },
+
+  sync: true,
+
+  fn: ({ requestedMove, currentState, playedBy }, exits) => {
+    try {
+      // Determine opponent (opponent is the other player)
+      const opponent = playedBy ? currentState.p0 : currentState.p1;
+
+      // Get the top two cards from the deck
+      const topTwoCards = currentState.deck.slice(0, 2);
+      const targetCard = opponent.points.find(({ id }) => id === requestedMove.targetId);
+      const playedCard = topTwoCards.find(({ id }) => id === requestedMove.cardId);
+
+      // Check if it's the player's turn
+      if (currentState.turn % 2 !== playedBy) {
+        throw new Error('game.snackbar.global.notYourTurn');
+      }
+
+      // Check if the game phase is RESOLVING_SEVEN
+      if (currentState.phase !== GamePhase.RESOLVING_SEVEN) {
+        throw new Error('game.snackbar.seven.pickAndPlay');
```

Comment:
I think this is likely the wrong error

Reply:
@Haviles04 it's a little silly but atm it's combined into a single error which handles the wrong phase in both directions:
```"pickAndPlay": "You can only play cards from the top of the deck while resolving a seven."```

idea being that it's an if and only if situation. It might make sense to pull out a separate wrongPhase move here but I don't feel strongly about it. Thoughts?

Reply:
Switched to `.wrongPhase`